tracer/src/Datadog.Tracer.Native/iast/iast_util.cpp

```diff
@@ -0,0 +1,1265 @@
+    {
+        clients.push_back(client);
+    }
+
+    void Event::Notify(void* data)
+    {
+        for (auto client : clients)
+        {
+            client(this, data);
+        }
+    }
+
+
+    //----------------------------------
+
+    template <typename Out>
+    void Split(const WSTRING& s, WCHAR delim, Out result)
+    {
+        size_t lpos = 0;
+        for (size_t i = 0; i < s.length(); i++)
+        {
+            if (s[i] == delim)
+            {
+                *(result++) = s.substr(lpos, (i - lpos));
+                lpos = i + 1;
+            }
+        }
+        *(result++) = s.substr(lpos);
+    }
+    std::vector<WSTRING> Split(const WSTRING& s, WCHAR delim)
+    {
+        std::vector<WSTRING> elems;
+        Split(s, delim, std::back_inserter(elems));
+        return elems;
+    }
+    std::vector<WSTRING> Split(const WSTRING& str, const WSTRING& delims)
+    {
+        std::vector<WSTRING> res;
+        std::size_t current, previous = 0;
+        current = str.find_first_of(delims);
+        WSTRING element;
+        while (current != WSTRING::npos)
+        {
+            element = str.substr(previous, current - previous);
+            if (element != EmptyWStr)
+            {
+                res.push_back(element);
+            }
+            previous = current + 1;
+            current = str.find_first_of(delims, previous);
+        }
+        element = str.substr(previous, current - previous);
+        if (element != EmptyWStr)
+        {
+            res.push_back(element);
+        }
+        return res;
+    }
+
+    template <typename Out>
+    void Split(const std::string& s, char delim, Out result)
+    {
+        size_t lpos = 0;
+        for (size_t i = 0; i < s.length(); i++)
+        {
+            if (s[i] == delim)
+            {
+                *(result++) = s.substr(lpos, (i - lpos));
+                lpos = i + 1;
+            }
+        }
+        *(result++) = s.substr(lpos);
+    }
+    std::vector<std::string> Split(const std::string& s, char delim)
+    {
+        std::vector<std::string> elems;
+        Split(s, delim, std::back_inserter(elems));
+        return elems;
+    }
+    std::vector<std::string> Split(const std::string& str, const std::string& delims)
+    {
+        std::vector<std::string> res;
+        std::size_t current, previous = 0;
+        current = str.find_first_of(delims);
+        std::string element;
+        while (current != std::string::npos)
+        {
+            element = str.substr(previous, current - previous);
+            if (element != "")
+            {
+                res.push_back(element);
+            }
+            previous = current + 1;
+            current = str.find_first_of(delims, previous);
+        }
+        element = str.substr(previous, current - previous);
+        if (element != "")
+        {
+            res.push_back(element);
+        }
+        return res;
+    }
+
+    int CharCount(const WSTRING& str, const WCHAR& lookup)
```

Comment:
instead of redefining this you can use `std::count` function on string https://en.cppreference.com/w/cpp/algorithm